pkg/controller/reconciler/annotations.go

```diff
@@ -84,7 +84,7 @@ type deploymentStatus struct {
 // This includes the configuration specified by the user, the hash of the configuration, and the status.
 func readAnnotations(deployment *appsv1.Deployment) (*deploymentAnnotations, error) {
 	if deployment.Annotations == nil {
-		return nil, nil
+		return &deploymentAnnotations{}, nil
```

Comment:
Before we start changing things, do we have a repro for this? Is it possible to add or tweak a test that will reproduce the failure.

Reply:
Asking because I'm not sure I understand the cause yet. Crashes like this are useful because they probably indicate a logic error.